langchain/utilities/bibtex.py

```diff
@@ -0,0 +1,177 @@
+        load_all_available_meta:
+          if True: the `metadata` of the loaded Documents gets all available meta info
+          if False: the `metadata` gets only the most informative fields.
+        doc_content_chars_max: the maximum number of characters in the content of a
+          Document 4000 by default
+        pdf_pattern: the regex pattern to find the pdf file name in the bibtex entry
+          r'[^:]+\.pdf' by default
+    """
+
+    bibtexparser_client: Any  #: :meta private:
+    bibtexparser_exceptions: Any  # :meta private:
+    load_max_docs: int = -1
+    load_all_available_meta: bool = False
+    doc_content_chars_max: int = 4000
+    pdf_pattern = r"[^:]+\.pdf"
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that the python package exists in environment."""
+        try:
+            import bibtexparser
+
+            values["load_file"] = bibtexparser.load
+            values["bibtexparser_exceptions"] = (
+                bibtexparser.bibtexexpression.BibtexExpression().ParseException,
+            )
+        except ImportError:
+            raise ValueError(
+                "Could not import bibtexparser python package. "
+                "Please install it with `pip install bibtexparser`."
+            )
+        return values
+
+    def _meta_str(self, entry: Any) -> str:
+        """
+        Return the meta information of an article in string format.
+        """
+        metadata = self._get_meta(entry)
+        return "\n".join([f"{key}: {value}" for key, value in metadata.items()])
+
+    def _get_metata(self, entry: Any) -> str:
+        meta = {
+            "ID": entry.get("ID", ""),
+            "Published": entry.get("year", "Published date unknown"),
+            "Title": entry.get("title", "Unknown"),
+            "Publication": entry.get("journal")
+            or entry.get("booktitle")
+            or "Unknown publication",
+            "Authors": entry.get("author", "Unknown authors"),
+            "Summary": entry.get("abstract", "No abstract available"),
+            "URL": entry.get("url", None) or f'https://doi.org/{entry.get("doi", "")}'
+            if entry.get("doi", None)
+            else "No URL available",
+        }
+        optional_fields = [
+            "annote",
+            "booktitle",
+            "editor",
+            "howpublished",
+            "journal",
+            "keywords",
+            "note",
+            "organization",
+            "publisher",
+            "school",
+            "series",
+            "type",
+            "doi",
+            "issn",
+            "isbn",
+        ]
+        if self.load_all_available_meta:
+            for field in optional_fields:
+                if field in entry:
+                    meta[field.capitalize()] = entry.get(field)
```

Comment:
Will break, but reminding to change to snake_case 

```suggestion
                    meta[field.snake_case()] = entry.get(field)
```